From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Tue, 25 Aug 2020 15:17:15 +0200
Subject: [PATCH] Use /lib for the dynamic linker on x86_64 by default

However, if '-gcc-toolchain /usr' is used, then assume that clang is
actually used as a host compiler and use the default /lib64 instead.

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 lib/Driver/ToolChains/Linux.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/Driver/ToolChains/Linux.cpp b/lib/Driver/ToolChains/Linux.cpp
index 6532c899492a..30026aea938a 100644
--- a/lib/Driver/ToolChains/Linux.cpp
+++ b/lib/Driver/ToolChains/Linux.cpp
@@ -633,8 +633,9 @@ std::string Linux::getDynamicLinker(const ArgList &Args) const {
     break;
   case llvm::Triple::x86_64: {
     bool X32 = Triple.getEnvironment() == llvm::Triple::GNUX32;
+    const Arg *A = Args.getLastArg(clang::driver::options::OPT_gcc_toolchain);
 
-    LibDir = X32 ? "libx32" : "lib64";
+    LibDir = X32 ? "libx32" : (A && A->getValue() == std::string("/usr")) ? "lib64" : "lib";
     Loader = X32 ? "ld-linux-x32.so.2" : "ld-linux-x86-64.so.2";
     break;
   }
